src/client/eppo-client.ts

```diff
@@ -393,16 +407,34 @@ export default class EppoClient implements IEppoClient {
     actions: Record<string, Attributes>, // TODO: ability to provide a set of actions with no context, or context broken out by numeric/categorical
     defaultValue: string,
   ): { variation: string; action: string | null } {
-    let variation = this.getStringAssignment(flagKey, subjectKey, subjectAttributes, defaultValue);
+    let variation = defaultValue;
     let action: string | null = null;
-    const banditKey = variation;
     try {
-      const banditParameters = this.banditConfigurationStore?.get(banditKey);
-      if (banditParameters && !Object.keys(actions ?? {}).length) {
-        // If it's a bandit, but we have no actions, just return default value
-        variation = defaultValue;
-      } else if (banditParameters) {
-        // For now, we use the shortcut of assuming if a variation value is the key of a known bandit, that is the bandit we want
+      const banditFlagAssociations = this.banditFlagConfigurationStore?.get(flagKey);
+      if (banditFlagAssociations && !Object.keys(actions).length) {
+        // No actions passed for a flag known to have an active bandit, so we just return the default values so that
+        // we don't log a variation or bandit assignment
+        return { variation, action };
```

Comment:
nit: slightly more explicit that we're returning a default value:
```suggestion
        return { variation: defaultVariation, action: null };
```


This also allows reducing the scope of `variation`/`action` variables. A related suggestion: in the catch branch, in graceful mode, it makes sense to return the default value explicitly as well. (So the reader doesn't have to eyeball where and how `variation`/`action` are changing through the function.)

Reply:
Ah yeah I can be explicit! We still need this scoped outside of the try-catch for the final return though